app/src/main/java/com/example/parkingsystemkotlin/mvp/model/ReservationModel.kt

```diff
@@ -37,14 +38,44 @@ class ReservationModel(private val reservationInformationDB: DataBase) : Reserva
 
     override fun saveReservation(securityCode: String, place: String) {
         if (::startDateAndTime.isInitialized && ::startDateAndTime.isInitialized && securityCode.isNotEmpty() && place.isNotEmpty()) {
-            reservationInformationDB.putReservationDB(startDateAndTime, finishDateAndTime, securityCode, place)
+            if (!isOverlap(place)) {
+                reservationInformationDB.putReservationDB(startDateAndTime, finishDateAndTime, securityCode, place)
+            }
         }
     }
 
+    private fun isOverlap(place: String): Boolean {
+        reservationInformationDB.getReservationsDB(place).forEach {
+            if (!(startDateAndTime.after(it.finishDateAndTime) || finishDateAndTime.before(it.startDateAndTime))) {
+                overlap = true
+                return true
+            }
+        }
+        return false
+    }
+
     override fun getReservation(place: String, securityCode: String): Reservation =
         reservationInformationDB.getReservationDB(place, securityCode)
 
     override fun setStartDateAndTime(startDateAndTime: Boolean) {
         this.isStartDateAndTime = startDateAndTime
     }
+
+    override fun thereWasOverlapOfReservations(): Boolean {
+        if (overlap) {
+            overlap = false
+            return true
+        }
+        return false
+    }
```

Comment:
This function follows the logic previous mentioned, only in this case you will check for the entire DB and not for a particular place.

Reply:
Whit this two changes you can deleted overlap variable.

Reply:
done